### ISSUE TITLE
which session of wetlab work?

### ISSUE THREAD

**Issue by zhang0730** [jkobject/scDataLoader#11]
error messsage:
ImportError: Python package for 'wetlab' is not installed.
If your package is on PyPI, run `pip install wetlab`
and my scdataloader version is 1.6.5
could you tell me which seesion of wetlab will work?

**Comment by jkobject** [comment]
Hello, sorry I came back from vacation and just seeing this. I reverted the latest version to 1.6.4 as there was problem otherwise. 
Let me know if this works with you now. I had never had this issue before can you tell me more about where it comes from and when?

**Comment by zhang0730** [comment]
welcome back! I hope you had a great vacation.
I had solve this problem by pip install "wetlab" "lamindb[bionty]==0.76.12" "bionty==0.52.0" --dry-run
and I have another question: if I wanted to process the scRNA-seq data from a dataset in GEO, which kind of input should be prepared for scdataloader; or put it another way, how can I process the scRNA-seq data from a GSE as a suitable input for scPRINT?

**Comment by jkobject** [comment]
Hello, there should be the "preprocess" function in scdataloader that allows you to transform the dataset in a suitable format with scdataloader. 

The goal is to put all of the datasets that scdataloader will use under the same format of gene names, cell types, diseases, ... and to filter for low-quality cells.

However, if you only have one dataset, you might not really need to use lamindb and care about this alignment of terms. So the pytorch datamodule might be too cumbersome. In this case, if you, for example, only care about weighted random sampling, you can do something like what I did in the tasks functions like task/denoising.py, where I use a pytorch dataloader, the collator, and SimpleAnnDataset. You can then use a weighted random sampler like I did in the datamodule class.

If I know more about your usecase, I might be better able to help.

Best,

**Comment by zhang0730** [comment]
thanks for instruction. I want to collect some scRNA-seq datasets with specific disease from different sources and manually annotate, as a input for pre-training. 
And I just finished watching your job interpretation video on youtube. what a great job!

**Comment by zhang0730** [comment]
thanks for instruction. I want to collect some scRNA-seq datasets with specific disease from different sources and manually annotate, as a input for pre-training. 
And I just finished watching your job interpretation video on youtube. what a great job!

**Comment by jkobject** [comment]
Ok, yes I think if you have many such files, working with lamindb and scdataloader will make your life much simpler. Maybe you know already but lamindb has toolkits to go from manual labelling of cells to ontological IDs. Once you have such a dataset of H5ADs, you can use scdataloader's preprocess class to preprocess them all to be the right format (with all of the available option of the class). Then you can do your pretraining. 

Let me know if/how you manage to make sure that there aren't duplicate datasets in when you do this?

**Comment by zhang0730** [comment]
yeah, because of the needs of specific diseases, I prefer to searching the pertinent article firstly and then to collecting their data provided, which can see the details of the data in the text. The sample size could be 500~1000. Welcome to give me some advice!

**Comment by jkobject** [comment]
I don't have much except to say you should look how CxG do it to get, annotate and validate datasets (they explain it in their doc). Also I have a script to get back the raw counts from normalized dataset if you want. 

Best,

**Comment by zhang0730** [comment]
Thank you for your suggestion. I will check out their documentation as you suggested.I am quite interested in the script you mentioned above. Could you please share more details or provide the script if possible?

Best regards,

**Comment by jkobject** [comment]
here it is:
`def unnormalize(df, is_root=False):
    # rescale (logp1 or sqroot)
    df = df**2 if is_root else (2**df) - 1
    r = np.array([i[i != 0].min() for k, i in df.iterrows()])
    df = (df.T / r).T.round()
    return df`

**Comment by zhang0730** [comment]
Thank you for sharing the script!

**Comment by zhang0730** [comment]
Thank you for sharing the script!
